File: src/data/courses/dsa/stacks/stack/stack-queue-deque/code/python/stack_example.py
class StackExample:
    def demonstrate(self):
        stack = []
        
        stack.append(5)
        stack.append(10)
        stack.append(15)
        
        print(f"Popped: {stack.pop()}")      # 15
        print(f"Top: {stack[-1]}")           # 10
        print(f"Size: {len(stack)}")         # 2
        print(f"Empty: {len(stack) == 0}")   # False

# Usage
example = StackExample()
example.demonstrate()

File: src/data/courses/dsa/stacks/stack/stack-queue-deque/code/cpp/StackExample.cpp
#include <iostream>
#include <stack>
using namespace std;

int main() {
    stack<int> st;
    
    st.push(5);
    st.push(10);
    st.push(15);
    
    cout << "Popped: " << st.top() << endl; // 15
    st.pop();
    cout << "Top: " << st.top() << endl;    // 10
    cout << "Size: " << st.size() << endl;  // 2
    cout << "Empty: " << st.empty() << endl; // 0 (false)
    
    return 0;
}
